File: spoj/2123-CANDY/solution_2123.cpp
#include <iostream>

using namespace std;

int sameCandyAmount(int candies[], int count);

int main()
{
    while (true)
    {
        int n;
        cin >> n;
        int candies[n];
        if (n == -1)
            break;
        for (int i = 0; i < n; i++)
        {
            int c;
            cin >> c;
            candies[i] = c;
        }
        cout << sameCandyAmount(candies, n) << "\n";
    }

    return 0;
}

int sameCandyAmount(int candies[], int count)
{
    int sum = 0;
    for (int i = 0; i < count; i++)
        sum += candies[i];

    if (sum % count != 0)
        return -1;

    int avg = sum / count;

    int movesNeeded = 0;
    for (int i = 0; i < count; i++)
        if (candies[i] < avg)
            movesNeeded += (avg - candies[i]);

    return movesNeeded;
}

File: spoj/2123-CANDY/solution_2123.py
def equal_candies(candies):
    total = sum(candies)

    if total % len(candies) != 0:
        return -1

    avg = total // len(candies)

    moves_required = sum([avg - x for x in candies if x < avg])

    return moves_required


if __name__ == "__main__":
    while True:
        n = int(input())
        if n == -1:
            break

        candies = []
        for x in range(n):
            candies.append(int(input()))

        print(equal_candies(candies))
